testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/injection/PostConstructInjectionTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @tpSubChapter Validation and @PostConstruct methods
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Regression tests for RESTEASY-2227
+ * @tpSince RESTEasy 3.6
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class PostConstructInjectionTest {
+
+   static Client client;
+
+   // deployment names
+   private static final String WAR_CDI_ON = "war_with_cdi_on";
+   private static final String WAR_CDI_OFF = "war_with_cdi_off";
+
+   /**
+    * Deployment with CDI activated
+    */
+   @Deployment(name = WAR_CDI_ON)
+   public static Archive<?> deployCdiOn() {
+      WebArchive war = TestUtil.prepareArchive(PostConstructInjectionTest.class.getSimpleName() + "_CDI_ON");
+      war.addAsWebInfResource(PostConstructInjectionTest.class.getPackage(), "PostConstructInjection_beans_cdi_on.xml", "beans.xml");
+      return TestUtil.finishContainerPrepare(war, null, PostConstructInjectionResource.class, PostConstructInjectionEJBInterceptorResource.class);
+   }
+
+   /**
+    * Deployment with CDI not activated
+    */
+   @Deployment(name = WAR_CDI_OFF)
+   public static Archive<?> deployCdiOff() {
+      WebArchive war = TestUtil.prepareArchive(PostConstructInjectionTest.class.getSimpleName() + "_CDI_OFF");
+      war.addAsWebInfResource(PostConstructInjectionTest.class.getPackage(), "PostConstructInjection_beans_cdi_off.xml", "beans.xml");
+      return TestUtil.finishContainerPrepare(war, null, PostConstructInjectionResource.class);
+   }
+
+   @BeforeClass
+   public static void init() {
+      client = ClientBuilder.newClient();
+   }
+
+   @AfterClass
+   public static void after() {
+      client.close();
+   }
+
+   private String generateURL(String jar, String path) {
+      return PortProviderUtil.generateURL(path, PostConstructInjectionTest.class.getSimpleName() + "_CDI_" + jar);
+   }
+
+   /**
+    * @tpTestDetails In an environment with managed beans, a @PostConstruct method on either an ordinary
+    *                resource or an EJB interceptor should execute before class and property validation is done.
+    * @tpSince RESTEasy 3.7.0
+    */
+   @Test
+   public void TestPostInjectCdiOn() throws Exception {
+      doTestPostInjectCdiOn("ON", "normal");
+      doTestPostInjectCdiOn("ON", "ejb");
+   }
+
+   /**
+    * @tpTestDetails In an environment without managed beans, a @PostConstruct method on a resource will not be called.
+    * @tpSince RESTEasy 3.7.0
+    */
+   @Test
+   public void TestPostInjectCdiOff() throws Exception {
+      Response response = client.target(generateURL("OFF", "/normal/get")).request().get();
+      Assert.assertEquals(200, response.getStatus());
+      Assert.assertEquals("ab", response.readEntity(String.class));
+      response.close();
+   }
+
+   void doTestPostInjectCdiOn(String cdi, String resource) {
+      Response response = client.target(generateURL(cdi, "/normal/get")).request().get();
```

Comment:
Seems this should use `resource` instead of the hardcoded `normal`

Reply:
@mackaypeter see https://issues.jboss.org/browse/RESTEASY-2264 and https://github.com/resteasy/Resteasy/pull/2036#issuecomment-497066576

Reply:
I see, thanks @spyrkob !